### PR TITLE
gorelease: build image suitable for CI

### PR DESCRIPTION
**Pull request by greut**
Create a `nancy:ci` Docker image that is suitable for Continuous integration such as Gitlab CI (where a sh is required)

This pull request makes the following changes:
* remove the `v` duplicate in the archive name in `--snapshot` mode.
* creates a second image for ci usage `nancy:ci`
* stop running the update/upgrade inside docker but pull the latest image

cc @bhamail / @DarthHater

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -77,3 +80,19 @@ dockers:
     - "sonatypecommunity/nancy:{{ .Tag }}"
     - "sonatypecommunity/nancy:v{{ .Major }}"
     - "sonatypecommunity/nancy:v{{ .Major }}.{{ .Minor }}"
```
Comment (zendern) [inline .goreleaser.yml]: If we are removing `v` from the archives should we also remove it here?? @DarthHater thoughts??

File: .goreleaser.yml [hunk 2]
```diff
@@ -77,3 +80,19 @@ dockers:
...
+    build_flag_templates:
+    - "--pull"
+    - "--label=author='DJ Schleen'"
+    - "--label=version={{ .Version }}"
+    - "--label=maintainer='sonatype-nexus-community'"
+    image_templates:
+    - "sonatypecommunity/nancy:ci"
+    - "sonatypecommunity/nancy:ci-{{ .Tag }}"
+    - "sonatypecommunity/nancy:ci-v{{ .Major }}"
+    - "sonatypecommunity/nancy:ci-v{{ .Major }}.{{ .Minor }}"
```
Comment (zendern) [inline .goreleaser.yml]: Same question about `v` or not to `v` here too.

File: .goreleaser.yml (hunk 1 above)
Comment (greut) [reply .goreleaser.yml]: I didn't see the issue here.

```
   • ARCHIVES         
      • skip archiving            binary=nancy.exe
      • skip archiving            binary=nancy
      • creating                  archive=dist/nancy-linux.386-vv0.1.5-next-2b79e57f21b0c370f22032c791da9c186f4c3da8.tar.gz
      • creating                  archive=dist/nancy-darwin.amd64-vv0.1.5-next-2b79e57f21b0c370f22032c791da9c186f4c3da8.tar.gz
      • skip archiving            binary=nancy
      • skip archiving            binary=nancy
      • creating                  archive=dist/nancy-darwin.386-vv0.1.5-next-2b79e57f21b0c370f22032c791da9c186f4c3da8.tar.gz
      • skip archiving            binary=nancy
      • creating                  archive=dist/nancy-windows.amd64-vv0.1.5-next-2b79e57f21b0c370f22032c791da9c186f4c3da8.zip
      • skip archiving            binary=nancy.exe
      • creating                  archive=dist/nancy-linux.amd64-vv0.1.5-next-2b79e57f21b0c370f22032c791da9c186f4c3da8.tar.gz
      • creating                  archive=dist/nancy-windows.386-vv0.1.5-next-2b79e57f21b0c370f22032c791da9c186f4c3da8.zip
```

File: .goreleaser.yml (hunk 2 above)
Comment (greut) [reply .goreleaser.yml]: `{{ .Version }}` shouldn't have the `v` though. Major, Minor doesnt' have it for sure.

https://goreleaser.com/customization/#Name%20Templates

File: .goreleaser.yml (hunk 1 above)
Comment (zendern) [reply .goreleaser.yml]: Cool I misunderstood what it was doing👍

Comment (greut) [reply .goreleaser.yml]: It seems to only do this in `--snapshot` mode. I've reverted that change.

Comment (greut) [reply .goreleaser.yml]: It was `{{ .Tag }}` vs `{{ .Version }}` in the snapshot section.

File: build/Dockerfile.alpine [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM alpine:latest
+
+RUN apk add --no-cache ca-certificates \
+    && rm -rf /var/cache/apk/*
+
+COPY nancy /usr/local/bin/nancy
+
+CMD ["nancy"]
```
Comment (DarthHater) [inline build/Dockerfile.alpine]: Shouldn't this be `ENTRYPOINT ["nancy"]`?

**Comment by greut** [comment]
The ci such as GitLab runner don't change the entry point by default. The cmd is set by each projects. A neutral entry point is thus required.

File: build/Dockerfile.alpine (hunk 3 above)
Comment (greut) [reply build/Dockerfile.alpine]: Usage example, https://gitlab.com/greut/eclint/-/jobs/471748010

**Comment by bhamail** [comment]
I attempted to resolve the conflicts and merged using PR #131. Please don't hesitate to create a new PR to fix any mess I made.

**Comment by greut** [comment]
@bhamail thanks mate! I'm looking forward to use it in GitLab CI soon.

**Pull request closed without merging**